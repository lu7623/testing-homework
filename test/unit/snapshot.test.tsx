import React from "react";
import renderer from "react-test-renderer";
import {Delivery} from "../../src/client/pages/Delivery"
import {Home} from "../../src/client/pages/Home"
import {Contacts} from "../../src/client/pages/Contacts"
import { Provider } from "react-redux";
import { ProductDetails } from "../../src/client/components/ProductDetails";
import { initState, LocalStorageMock } from "../helper";
import { Product } from "../../src/common/types";
import { Catalog } from "../../src/client/pages/Catalog";
import { Cart } from "../../src/client/pages/Cart";

afterEach(() => {
  jest.clearAllMocks();
});

describe("Страницы отображаются верно", () => {
  it("Страница доставки сохраняет статическое содержание", () => {
    const domTree = renderer.create(<Delivery />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Главная страница сохраняет статическое содержание", () => {
    const domTree = renderer.create(<Home />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Страница контактов сохраняет статическое содержание", () => {
    const domTree = renderer.create(<Contacts />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Страница продукта отображается верно", () => {
    const init = initState();
    const mockProduct: Product = {
        color: "Blue",
        description: "Really Incredible kogtetochka for Sphynx",
        material: "cotton",
        name: "Best kogtetochka",
        id: 123,
        price: 200,
      };
    const domTree = renderer.create(<Provider store={init}>
        <ProductDetails product={mockProduct} />
      </Provider>).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Страница каталога отображается верно", () => {
   
  const init = initState()
    const domTree = renderer.create(<Provider store={init}>
        <Catalog />
      </Provider>).toJSON();
    expect(domTree).toMatchSnapshot();
  });
  it("Страница корзины отображается верно", () => {
    global.localStorage = new LocalStorageMock();
    localStorage.setItem('example-store-cart', JSON.stringify({
      "0": {
          "name": "Solid kogtetochka",
          "count": 3,
          "price": 200
      },
      "1": {
          "name": "Luxury kogtetochka",
          "count": 2,
          "price": 1300
      }
  }))
   const init = initState()
    const domTree = renderer.create(<Provider store={init}>
        <Cart/>
      </Provider>).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});