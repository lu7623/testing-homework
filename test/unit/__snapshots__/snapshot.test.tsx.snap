// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страницы отображаются верно Главная страница сохраняет статическое содержание 1`] = `
<div
  className="Home"
>
  <div
    className="row"
  >
    <div
      className="col bg-orange text-white py-4 bg-opacity-75"
    >
      <p
        className="display-3"
      >
        Welcome to Kogtetochka store!
      </p>
      <p
        className="lead"
      >
        We have a large assortment of scratching posts!
      </p>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Stability
      </h1>
      <p
        className="lead"
      >
        Our scratching posts are crafted with precision and designed for unparalleled stability. Made from high-quality materials, they provide a sturdy platform for your cat's scratching needs.
      </p>
    </div>
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Comfort
      </h1>
      <p
        className="lead"
      >
        Pamper your feline friend with the luxurious comfort of our scratching posts. Covered in soft, plush fabric, they offer a cozy retreat for your cat to relax and unwind.
      </p>
    </div>
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Design
      </h1>
      <p
        className="lead"
      >
        Engage your cat's natural instincts and keep them entertained for hours with our interactive scratching posts. Featuring built-in toys and enticing textures, they stimulate your cat's senses and encourage active play.
      </p>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-12py-3"
    >
      <p
        className="fs-1"
      >
        Empower Your Coding Journey with Every Scratch – Get Your Paws on Our Purr-fect Scratchers Today!
      </p>
    </div>
  </div>
</div>
`;

exports[`Страницы отображаются верно Страница доставки сохраняет статическое содержание 1`] = `
<div
  className="Delivery"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Delivery
      </h1>
      <p>
        Swift and Secure Delivery: Experience the convenience of hassle-free shipping with our scratchers. We understand the excitement of receiving your new cat furniture, so we prioritize swift delivery to your doorstep. Rest assured, your order is handled with care every step of the way, ensuring it arrives safely and securely.
      </p>
      <img
        className="Image w-25 mb-4"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
      />
      <p>
        Track Your Package with Ease: Stay informed and in control of your delivery with our easy-to-use tracking system. From the moment your order is placed to the minute it reaches your home, you can monitor its journey in real-time. No more guessing games – know exactly when to expect your package and plan accordingly.
      </p>
      <p>
        Customer Satisfaction Guaranteed: Your satisfaction is our top priority, which is why we go above and beyond to provide exceptional delivery service. If you have any questions or concerns about your shipment, our dedicated customer support team is here to assist you every step of the way. Trust us to deliver not only your scratcher but also peace of mind.
      </p>
    </div>
  </div>
</div>
`;

exports[`Страницы отображаются верно Страница каталога отображается верно 1`] = `
<div
  className="Catalog"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Catalog
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    LOADING
  </div>
</div>
`;

exports[`Страницы отображаются верно Страница контактов сохраняет статическое содержание 1`] = `
<div
  className="Contacts"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Contacts
      </h1>
      <p>
        Have a question about our scratchers or need help placing an order? Don't hesitate to reach out to us! Our dedicated team is here to provide you with top-notch service and support.
      </p>
      <p>
        Our friendly representatives are available during business hours to assist you with any inquiries you may have.
      </p>
      <p>
        At our store, customer satisfaction is our priority, and we're committed to ensuring you have a smooth and enjoyable shopping experience. Reach out to us today – we're here to help make your cat's scratching dreams a reality!
      </p>
    </div>
  </div>
</div>
`;

exports[`Страницы отображаются верно Страница корзины отображается верно 1`] = `
<div
  className="Cart"
>
  <div
    className="row mb-4"
  >
    <div
      className="col"
    >
      <h1>
        Shopping cart
      </h1>
      <table
        className="Cart-Table table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              Product
            </th>
            <th
              scope="col"
            >
              Price
            </th>
            <th
              scope="col"
            >
              Count
            </th>
            <th
              scope="col"
            >
              Total
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="0"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              1
            </th>
            <td
              className="Cart-Name"
            >
              Solid kogtetochka
            </td>
            <td
              className="Cart-Price"
            >
              $
              200
            </td>
            <td
              className="Cart-Count"
            >
              3
            </td>
            <td
              className="Cart-Total"
            >
              $
              600
            </td>
          </tr>
          <tr
            data-testid="1"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              2
            </th>
            <td
              className="Cart-Name"
            >
              Luxury kogtetochka
            </td>
            <td
              className="Cart-Price"
            >
              $
              1300
            </td>
            <td
              className="Cart-Count"
            >
              2
            </td>
            <td
              className="Cart-Total"
            >
              $
              2600
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colSpan={4}
            >
              Order price:
            </td>
            <td
              className="Cart-OrderPrice"
            >
              $
              3200
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-6"
    >
      <button
        className="Cart-Clear btn btn-outline-secondary"
        onClick={[Function]}
      >
        Clear shopping cart
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-sm-8 col-md-6 col-lg-4"
    >
      <h2>
        Сheckout
      </h2>
      <div
        className="Form"
      >
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-name"
          >
            Name
          </label>
          <input
            autoComplete="off"
            className="Form-Field Form-Field_type_name form-control"
            disabled={false}
            id="f-name"
            onChange={[Function]}
            type="text"
          />
          <div
            className="invalid-feedback"
          >
            Please provide your name
          </div>
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-phone"
          >
            Phone
          </label>
          <input
            className="Form-Field Form-Field_type_phone form-control"
            disabled={false}
            id="f-phone"
            onChange={[Function]}
            type="text"
          />
          <div
            className="invalid-feedback"
          >
            Please provide a valid phone
          </div>
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-address"
          >
            Address
          </label>
          <textarea
            className="Form-Field Form-Field_type_address form-control"
            disabled={false}
            id="f-address"
            onChange={[Function]}
            rows={3}
          />
          <div
            className="invalid-feedback"
          >
            Please provide a valid address
          </div>
        </div>
        <button
          className="Form-Submit btn btn-primary"
          disabled={false}
          onClick={[Function]}
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Страницы отображаются верно Страница продукта отображается верно 1`] = `
<div
  className="ProductDetails row"
>
  <div
    className="col-12 col-sm-5 col-lg-4"
  >
    <img
      className="Image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
  </div>
  <div
    className="col-12 col-sm-7 col-lg-6"
  >
    <h1
      className="ProductDetails-Name"
    >
      Best kogtetochka
    </h1>
    <p
      className="ProductDetails-Description"
    >
      Really Incredible kogtetochka for Sphynx
    </p>
    <p
      className="ProductDetails-Price fs-3"
    >
      $
      200
    </p>
    <p>
      <button
        className="ProductDetails-AddToCart btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </p>
    <dl>
      <dt>
        Color
      </dt>
      <dd
        className="ProductDetails-Color text-capitalize"
      >
        Blue
      </dd>
      <dt>
        Material
      </dt>
      <dd
        className="ProductDetails-Material text-capitalize"
      >
        cotton
      </dd>
    </dl>
  </div>
</div>
`;
